worlds/faxanadu/Rules.py

```diff
@@ -0,0 +1,94 @@
+from typing import TYPE_CHECKING
+from worlds.generic.Rules import set_rule
+
+if TYPE_CHECKING:
+    from . import FaxanaduWorld
+
+
+def can_buy_in_eolis(state, player):
+    # Sword or Deluge so we can farm for gold.
+    # Ring of Elf so we can get 1500 from the King.
+    return state.has("Progressive Sword", player, 1) or \
+           state.has("Deluge", player, 1) or \
+           state.has("Ring of Elf", player, 1)
+
+
+def has_any_magic(state, player):
+    return state.has("Deluge", player, 1) or \
+           state.has("Thunder", player, 1) or \
+           state.has("Fire", player, 1) or \
+           state.has("Death", player, 1) or \
+           state.has("Tilte", player, 1)
+
+
+def set_rules(faxanadu_world: "FaxanaduWorld"):
+    player = faxanadu_world.player
+    world = faxanadu_world.multiworld
```

Comment:
```suggestion
    multiworld = faxanadu_world.multiworld
```
See #2749. You'll have to adjust the uses of this, so don't directly commit this suggestion.

Reply:
`multi_world` instead would be better?

Reply:
`multiworld` is already being used in Regions.py. It's up to you which to use, but I'd say you should make them consistent.